autopilot/agent.go

```diff
@@ -470,23 +529,34 @@ func (a *Agent) controller(startingBalance btcutil.Amount) {
 							directive.ChanAmt, err)
 
 						// As the attempt failed, we'll
-						// clear it from the set of
-						// pending channels.
+						// disconnect the peer and clear
+						// it from the set of pending
+						// channels.
+						err = a.cfg.DisconnectPeer(
```

Comment:
Should also avoid disconnecting here if we were already connected.

Reply:
Fixed.